wpilibc/src/main/native/include/frc/controller/ControlAffinePlantInversionFeedforward.h

```diff
@@ -0,0 +1,185 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <array>
+#include <functional>
+
+#include <Eigen/Core>
+#include <units/units.h>
+
+#include "frc/system/NumericalJacobian.h"
+
+namespace frc {
+
+template <int N>
+using Vector = Eigen::Matrix<double, N, 1>;
+
+/**
+ * Constructs a control-affine plant inversion model-based feedforward from
+ * given model dynamics.
+ *
+ * <p>If given the vector valued function as f(x, u) where x is the state
+ * vector and u is the input vector, the B matrix(continuous input matrix)
```

Comment:
Put spaces between words and parentheticals (e.g., `matrix (continuous input matrix)`).